app/src/main/java/com/islamey/hamza/wazaker/ui/HomeFragment/HomeFragment.kt

```diff
@@ -60,50 +64,45 @@ class HomeFragment : Fragment() {
             findNavController().navigate(action)
         }
 
-        binding.fortyHadithbtn.setOnClickListener {
-            val action = HomeFragmentDirections.actionMainFragmentToFortyHadithListFragment()
-            findNavController().navigate(action)
-        }
-
         binding.counterBtn.setOnClickListener {
             val action = HomeFragmentDirections.actionMainFragmentToCounterFragment()
             findNavController().navigate(action)
         }
+    }
 
-        binding.qiblaBtn.setOnClickListener {
-            openQiblaActivity()
+    private fun observeResponse() {
+        viewLifecycleOwner.lifecycleScope.launchWhenStarted {
+            hijriViewModel.hijriDate.collect { state ->
+                when (state) {
+                    is DataState.Idle -> {
+                        binding.hijriCardView.isVisible = false
+                    }
+                    is DataState.Loading -> {
+                        binding.hijriCardView.isVisible = false
+                    }
+                    is DataState.Success -> {
+                        binding.hijriCardView.isVisible = true
+                        updateUi(state.data)
+                    }
+                    is DataState.Error -> {
+                        Log.e("hamzaError", state.message)
+                    }
+                }
+            }
         }
 
-    }
+        viewLifecycleOwner.lifecycleScope.launch {
+            counterViewModel.totalCounts.collect { totalCounts ->
+                binding.totalZekercounts.text = getString(R.string.totalzeker) + " " + totalCounts
```

Comment:
search on how to use get string from resources with placeholder

Reply:
mean like this ? 

```kotlin
<string name="totalzeker">Total zeker:  %s!</string>

String totalzeker = context.getString(R.string.totalzeker, totalZeker);
```

Reply:
exactly